### ISSUE TITLE
smb: Backups no longer working (1.2.1 patch regression)

### ISSUE THREAD

**Issue by sophie-h** [borgbackup/borg#6786]
## Have you checked borgbackup docs, FAQ, and open Github issues?

No

## Is this a BUG / ISSUE report or a QUESTION?

BUG

#### Your borg version (borg -V).

1.2.1

#### Hardware / network configuration, and filesystems used.

Debian smb server / gvfs

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

`borg create ...` on `/run/user/1000/gvfs/smb-share:server`

## Describe the problem you're observing.

```
    os.chmod(self.tmp_fname, mode=0o666 & ~ umask)
OSError: [Errno 95] Operation not supported: '/run/user/1000/gvfs/smb-share:server=…/test/nonce-wgo81cle.tmp'

Platform: Linux desktop 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64
Linux: Unknown Linux  
Borg: 1.2.1  Python: CPython 3.9.9 msgpack: 1.0.4 fuse: pyfuse3 3.2.1 [pyfuse3,llfuse]
```

Looks like this is a problem because SMB just doesn't support chmod (at least I can't do it via terminal with gvfs)? 0f9d88bf2e1f2d14be8fa30810948ef0e4107f42

Original issue is https://gitlab.gnome.org/World/pika-backup/-/issues/241

**Comment by ThomasWaldmann** [comment]
Yeah, 0f9d88b is the changeset that introduced this to fix another issue. Guess we need a try / except there.

If the chmod does not work, it is usually no problem except that the mode of the file looks like the umask was not respected. As long as the stuff is owned by the borg user, the owner permissions are enough.

**Comment by ThomasWaldmann** [comment]
There has been another attempt to fix this, please test the code from PR #6935 #6936 #6940.

See also the issue in the python issue tracker: https://github.com/python/cpython/issues/95658
